src/imarker_end_effector.cpp

```diff
@@ -141,22 +144,23 @@ bool IMarkerEndEffector::setPoseFromRobotState()
   return true;
 }
 
-void IMarkerEndEffector::iMarkerCallback(const visualization_msgs::InteractiveMarkerFeedbackConstPtr& feedback)
+void IMarkerEndEffector::iMarkerCallback(
+    const visualization_msgs::msg::InteractiveMarkerFeedback::ConstSharedPtr& feedback)
 {
-  if (feedback->event_type == visualization_msgs::InteractiveMarkerFeedback::MOUSE_UP)
+  if (feedback->event_type == visualization_msgs::msg::InteractiveMarkerFeedback::MOUSE_UP)
   {
     // Save pose to file if its been long enough
     double save_every_sec = 0.1;
-    if (time_since_last_save_ < ros::Time::now() - ros::Duration(save_every_sec))
+    if (time_since_last_save_ < rclcpp::Clock(RCL_ROS_TIME).now() - rclcpp::Duration(save_every_sec))
```

Comment:
```suggestion
    if (time_since_last_save_ < rclcpp::Clock(RCL_ROS_TIME).now() - rclcpp::Duration::from_seconds(save_every_sec))
```

Reply:
Also, I think we should have a clock as a member variable rather than constructing it on the fly on each call to this function